### DISCUSSION TITLE
Odd behavior parsing certain directory paths with cmgen command line

### DISCUSSION THREAD

**Discussion by EnviralDesign** [google/filament#5121] [Q&A]
This may very well be a windows command line quirk.. but I'm not sure.

If I run a bat file that looks like this, everything works fine:
```
"C:\tmp\filament\bin\cmgen.exe" ^
--ibl-dfg="C:\tmp\filament\bin\out&put\ArtistWorkShop\dfg.dds" ^
--sh-irradiance ^
--ibl-dfg-multiscatter ^
--sh-shader ^
--sh-output="C:\tmp\filament\bin\out&put\ArtistWorkShop\sh.txt" ^
--ibl-ld="C:\tmp\filament\bin\out&put\ArtistWorkShop" ^
--format=dds ^
"C:\tmp\filament\bin\invalid^&Folder\ArtistWorkShop.hdr"
```

however, if I simply remove the end of line carrot symbols, and reduce the command to one line like this:
```
"C:\tmp\filament\bin\cmgen.exe" --ibl-dfg="C:\tmp\filament\bin\out&put\ArtistWorkShop\dfg.dds" --sh-irradiance --ibl-dfg-multiscatter --sh-shader --sh-output="C:\tmp\filament\bin\out&put\ArtistWorkShop\sh.txt" --ibl-ld="C:\tmp\filament\bin\out&put\ArtistWorkShop" ^
--format=dds "C:\tmp\filament\bin\invalid^&Folder\ArtistWorkShop.hdr"
```
A weird thing happens - cmgen creates the dfg.dds successfully, creates an sh.txt (but with all zeroes inside) and also creates the sub folder containing all the cubemap mip level tiles, but all the tiles are black when viewed.

I do believe my issue has something to do with the ampersand in the name of the folder (out&put) but I find it really odd that it sort of works, instead of totally breaks.

I have to escape the ampersand in the last argument otherwise I get a "file could not be found" type of error. 
```
"C:\tmp\filament\bin\invalid^&Folder\ArtistWorkShop.hdr"
```

Any ideas on if this is a path parsing issue on my end, or if this is a bug in the way the command line stuff is parsed?
Thank you!

**Comment by bejado** [comment]
I think you need to be consistent and escape all the ampersands, not just the last one in the output file name. Ampersands on Windows are used to [separate commands](https://stackoverflow.com/questions/1327431/how-do-i-escape-ampersands-in-batch-files). Not sure why the behavior is different between the multi-line and single line command though.

**Reply by EnviralDesign** [reply]
I did try that, but then the command broke sooner, not finding those paths.
I went ahead and just changed that folder on my computer to not have an ampersand, was curious if I was missing anything else that might be obvious.
